modules/phase_field/examples/multiphase/GrandPotential3Phase.i

```diff
@@ -0,0 +1,337 @@
+  [../]
+  [./coupled_etab0dot]
+    type = CoupledSwitchingTimeDerivative
+    variable = w
+    v = etab0
+    Fj_names = 'rhoa rhob rhod'
+    hj_names = 'ha   hb   hd'
+    args = 'etaa0 etab0 etad0'
+  [../]
+  [./coupled_etad0dot]
+    type = CoupledSwitchingTimeDerivative
+    variable = w
+    v = etad0
+    Fj_names = 'rhoa rhob rhod'
+    hj_names = 'ha   hb   hd'
+    args = 'etaa0 etab0 etad0'
+  [../]
+[]
+
+[Materials]
+  [./ha_test]
+    type = SwitchingFunctionMultiPhaseMaterial
+    h_name = ha
+    all_etas = 'etaa0 etab0 etad0'
+    phase_etas = 'etaa0'
+  [../]
+  [./hb_test]
+    type = SwitchingFunctionMultiPhaseMaterial
+    h_name = hb
+    all_etas = 'etaa0 etab0 etad0'
+    phase_etas = 'etab0'
+  [../]
+  [./hd_test]
+    type = SwitchingFunctionMultiPhaseMaterial
+    h_name = hd
+    all_etas = 'etaa0 etab0 etad0'
+    phase_etas = 'etad0'
+  [../]
+  [./omegaa]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = omegaa
+    material_property_names = 'Vm ka caeq'
+    function = '-0.5*w^2/Vm^2/ka-w/Vm*caeq'
+    derivative_order = 2
+  [../]
+  [./omegab]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = omegab
+    material_property_names = 'Vm kb cbeq'
+    function = '-0.5*w^2/Vm^2/kb-w/Vm*cbeq'
+    derivative_order = 2
+  [../]
+  [./omegad]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = omegad
+    material_property_names = 'Vm kd cdeq'
+    function = '-0.5*w^2/Vm^2/kd-w/Vm*cdeq'
+    derivative_order = 2
+  [../]
+  [./rhoa]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = rhoa
+    material_property_names = 'Vm ka caeq'
+    function = 'w/Vm^2/ka + caeq/Vm'
+    derivative_order = 2
+  [../]
+  [./rhob]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = rhob
+    material_property_names = 'Vm kb cbeq'
+    function = 'w/Vm^2/kb + cbeq/Vm'
+    derivative_order = 2
+  [../]
+  [./rhod]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = rhod
+    material_property_names = 'Vm kd cdeq'
+    function = 'w/Vm^2/kd + cdeq/Vm'
+    derivative_order = 2
+  [../]
+  [./c]
+    type = ParsedMaterial
+    material_property_names = 'Vm rhoa rhob rhod ha hb hd'
+    function = 'Vm * (ha * rhoa + hb * rhob + hd * rhod)'
+    f_name = c
+  [../]
+  [./const]
+    type = GenericConstantMaterial
+    prop_names =  'kappa_c  kappa   L   D    Vm   ka    caeq kb    cbeq  kd    cdeq  gab gad gbd  mu  tgrad_corr_mult'
+    prop_values = '0        1       1.0 1.0  1.0  10.0  0.1  10.0  0.9   10.0  0.5   1.5 1.5 1.5  1.0 0.0'
+  [../]
+  [./Mobility]
+    type = DerivativeParsedMaterial
+    f_name = Dchi
+    material_property_names = 'D chi'
+    function = 'D*chi'
+    derivative_order = 2
+  [../]
+  [./chi]
+    type = DerivativeParsedMaterial
+    f_name = chi
+    material_property_names = 'Vm ha ka hb kb hd kd'
+    function = '(ha/ka + hb/kb + hd/kd) / Vm^2'
+    derivative_order = 2
+  [../]
```

Comment:
Which derivatives do you expect should be taken here? (because there won't be any unless you add args and specify the variable dependencies of your material properties)

Reply:
I guess this should probably be

```
  [./chi]
    type = DerivativeParsedMaterial
    f_name = chi
    material_property_names = 'Vm ha(etaa0,etab0,etad0) ka hb(etaa0,etab0,etad0) kb hd(etaa0,etab0,etad0) kd'
    function = '(ha/ka + hb/kb + hd/kd) / Vm^2'
    args = 'etaa0 etab0 etad0'
    derivative_order = 2
  [../]
```

Reply:
right thanks